app/util/api_version_handler.py

```diff
@@ -0,0 +1,78 @@
+import re
+
+
+class APIVersionHandler(object):
+    # List of all versions of the API.
+    _versions = [
+        1,
+        2
+    ]
+
+    @classmethod
+    def resolve_version(cls, header: str, uri: str):
+        """
+        Gets the respective version of the API relative to the request header and URI.
+        :param header: The value of the header which to search for version type (Content-Type/Accept).
+        :type header: str
+        :param uri: The URI from the request being checked.
+        :type uri: str
+        :rtype: int
+        """
+        matches = re.match(r'.*(?:application/vnd.clusterrunner.v(\d+)\+json).*', header, re.I)
```

Comment:
I find this logic hard to read since it's split across 5 methods (`resolve_versions`, `_get`, `_get_default`, `get_first`, `get_last`). How do you feel about condensing the first 3 of those into 1 method? (Ideally i'd like to condense the last 2 also, but they're public methods)
```python
def resolve_version(header, uri):
  header_version = None
  try:
    match =  re.match(r'.*(?:application/vnd.clusterrunner.v(\d+)\+json).*', 
                       header, re.I)
    header_version = int(match.group(0))
  except (AttributeError, ValueError):
    pass

  uri_version = next(part for part in uri.split('/') if part != '', None)
  requested_version = header_version or uri_version
  return requested_version if requested_version in cls._versions else DEFAULT_VERSION 
```

I _think_ that's the same logic as what you've laid out